tests/framework/matcher/phase_test.go

```diff
@@ -0,0 +1,30 @@
+package matcher
+
+import (
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+)
+
+var _ = Describe("Matcher", func() {
+
+	var runningPod = &v1.Pod{
+		Status: v1.PodStatus{
+			Phase: v1.PodRunning,
+		},
+	}
+
+	table.DescribeTable("should", func(exptectedPhase interface{}, pod *v1.Pod, match bool) {
+		success, err := BeInPhase(exptectedPhase).Match(pod)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(success).To(Equal(match))
+		Expect(BeInPhase(exptectedPhase).FailureMessage(pod)).ToNot(BeEmpty())
+		Expect(BeInPhase(exptectedPhase).NegatedFailureMessage(pod)).ToNot(BeEmpty())
+	},
+		table.Entry("with expected phase as PodPhase match the pod phase", v1.PodRunning, runningPod, true),
+		table.Entry("with expected phase as string match the pod phase", "Running", runningPod, true),
+		table.Entry("cope with a nil pod", v1.PodRunning, nil, false),
+		table.Entry("with expected phase not match the pod phase", "Succeeded", runningPod, false),
```

Comment:
Maybe we could add an additional case for unexisting actual phase (ie. `runtime.DefaultUnstructuredConverter.ToUnstructured` or `unstructured.NestedString` returning an error), wdyt?

Reply:
Done.